### DISCUSSION TITLE
Generate open gate message and share on social media

### DISCUSSION THREAD

**Discussion by nowaycs** [spartan-protocol/SpartanProtocol-DAppV2#911] [Ideas]
It would be nice to have a function that auto generates a message + screenshot of a capped/almost capped pool and be able to share this message on popular social media platforms like Twitter, Instagram and so on.

Or at least add a auto generated text with pool info and open the gates message to manually paste in social media platforms.

What do you think?

**Comment by SamusElderg** [comment]
You are welcome to add this as an issue (despite the misleading title 'issues' its where features/tasks go too) and see if someone puts up their hand to work on it.

Id suggest this general plan of stages:

**Stage 1 (Functional):**
- Button below capped pool item (if its red) in the pools-table && grid view
- The button launches a modal component
- In the modal have the suggested tweet text with a copy button and some general info about posting it on Twitter and tagging which/what
- Show an image of the capped pool in a layout/dimensions suitable to Twitter as an image-share (for the user to manually screenshot and share in the tweet)

**Stage 2 (UX Improvement):**
- Add a 'twitter' icon button next to the copy button to open Twitter in the user's browser with the copied text pre-entered for them
- Bonus but not critical: See if you can automate an image into the user's clipboard (or handed to Twitter API somehow) of the capped pool and implement that into the Tweet content too (this might be a bit problematic especially across browsers/OS/security-settings in terms of compatibility)
- Think outside the box if the 'take a screenshot for the user' approach is problematic, ie: might alternatively be able to render the 'thing to be screenshotted' as an image instead of HTML (in the modal) for a right-click -> save/copy
- Explore the options for better UX for the screenshot but dont stress too much if there is no clean solution, this is already an improvement in UX for Tweeting out capped pools even at stage 1
